# Documentação de Limpeza das Planilhas

A documentação a seguir descreve as etapas realizadas no processo de limpeza das planilhas, previo à carga de dados no sistema SAP.


# Tabela OCRB - Dados Bancários



A aba OCRB armazena os dados bancários dos parceiros de negócios, necessários para a execução de transações financeiras, tanto para clientes quanto para fornecedores. As informações incluem:

- **Código do banco e Agência:** Identificação da instituição bancária.
- **Número da conta:** Conta bancária do parceiro.
- **Nome da conta:** Titular da conta.
- **Conta contábil:** Conta usada para registrar as transações financeiras no sistema.

Esses dados garantem que todas as operações financeiras, como pagamentos e reembolsos, sejam realizadas de forma correta e eficiente.

## Alterações Realizadas

- Adição dos nomes dos bancos
- Separação do dígito da conta quando informado

Em algumas linhas falta o **dígito da conta** *(coluna UserNo2)*, sendo essas:

In [ ]:
import pandas as pd

file_path = '/content/Filtered_UserNo2Data_Updated.xlsx'

df = pd.ExcelFile(file_path)

# Exibir as planilhas disponíveis no arquivo Excel
print(df.sheet_names)

# Ler uma planilha específica (por exemplo, a primeira planilha)
df_sheet = df.parse(df.sheet_names[0])

# Visualizar os primeiros registros do DataFrame
df_sheet.head(100)


['Sheet1']


ParentKey County  BankCode  Branch  UserNo1 AccountNo  UserNo2  \
0      C0462     BR       237  2224.0      NaN       657      NaN   
1      C0471     BR       341  8777.0      NaN     21484      NaN   
2      C0473     BR       341   910.0      NaN     13198      NaN   
3      C0476     BR       341   421.0      NaN     39585      NaN   
4      C0477     BR       341  1633.0      NaN     16201      NaN   
..       ...    ...       ...     ...      ...       ...      ...   
95     F1067     BR       341  1042.0      NaN     00762      NaN   
96     F1068     BR       341  1669.0      NaN     45091      NaN   
97     F1069     BR       341  6315.0      NaN     22716      NaN   
98     F1070     BR       341  3748.0      NaN     11096      NaN   
99     F1070     BR       260     1.0      NaN  29392235      NaN   

       AccountName  Unnamed: 8  
0   Banco Bradesco         NaN  
1    Itaú Unibanco         NaN  
2    Itaú Unibanco         NaN  
3    Itaú Unibanco         NaN  
4    Itaú Unibanco         NaN  
..             ...         ...  
95   Itaú Unibanco         NaN  
96   Itaú Unibanco         NaN  
97   Itaú Unibanco         NaN  
98   Itaú Unibanco         NaN  
99   Nu Pagamentos         NaN  

[100 rows x 9 columns]

# Tabela OCRD - Dados Gerais de Parceiros de Negócio

A aba OCRD reúne os dados gerais dos parceiros de negócios, sejam eles clientes ou fornecedores. Esta planilha é fundamental para o cadastro inicial dos parceiros no SAP Business One, pois contém informações como:

- **Código do parceiro de negócio:** Código alfanumérico que identifica o parceiro dentro do sistema.
- **Razão social:** Nome registrado do parceiro de negócio.
- **Tipo de Parceiro de Negócio:** Identifica se o parceiro é um cliente ou fornecedor.
- **Grupo do Parceiro:** Categoria do parceiro dentro do sistema.
- **Contatos:** Telefone, e-mail e outros dados de contato essenciais.

Esses dados são cruciais para o gerenciamento das relações comerciais e logísticas da empresa.

## Alterações Realizadas

- Normatização das strings
- `Coluna CardType`: Alteração dos valores para as variáveis corretas da base do SAP pelo Sheets
- `Coluna GroupCode`: Atualização dos valores com os códigos correspondentes na base do SAP pelo Sheets
- Separação do **DDD** *(Coluna Phone2)* do **Telefone** *(coluna Phone1)*
- Adição de hífen nos números de telefone e celular
- Organização manual de e-mails, websites e outras informações opcionais que estavam em locais equivocados
- Substituição dos nomes *SalesPersonCode* pelos códigos disponibilizados no BBP

Além disso, algumas linhas possuem o telefone, mas falta o **DDD** *(coluna Phone2)*, sendo essas:

In [ ]:
file_path = '/content/Filtered_PhoneData.xlsx'

df = pd.ExcelFile(file_path)

# Exibir as planilhas disponíveis no arquivo Excel
print(df.sheet_names)

# Ler uma planilha específica (por exemplo, a primeira planilha)
df_sheet = df.parse(df.sheet_names[0])

# Visualizar os primeiros registros do DataFrame
df_sheet.head(10)


['Sheet1']


CardCode                                           CardName  \
0  CardCode                                           CardName   
1    C00001             EMPRESA BRASILEIRA DE ENGENHARIA CIVIL   
2     F1048                                   V4 Company Ltda    
3  L00764                      Porto Seguro Imoveis S C Ltda     
4     C0303                                   A L A LIMA-EPP     
5     C0051        NTX Comércio, Consultoria e Serviços Ltda.    
6     C0052                                G2 Tecnologia ltda    
7     C0053     G2 Analise e Desenvolvimento de Sistemas ltda    
8     C0065                                            Quintec   
9     C0070  Lavoro Sano Consultoria e Gestão em Saúde do T...   

                                 CardForeignName   CardType      GroupCode  \
0                                      CardFName   CardType      GroupCode   
1                               ENGENHARIA CIVIL  cCustomer     ENGENHARIA   
2                                V4 Company Ltda  cSupplier  Fornecedores    
3                                            NaN  cSupplier            114   
4                                   Papel Decor        cLid            100   
5                                            NTX  cCustomer            165   
6                                         G2 Tec  cCustomer            178   
7                                         G2 Ana  cCustomer            178   
8  Quintec Brasil Serviços de Informática Ltda    cCustomer            102   
9                                            NaN  cCustomer            178   

           Phone1  Phone2      Cellular          Fax  \
0               1     NaN           NaN          Fax   
1       4141-8828     NaN  1194141-8828          NaN   
2  0800 608 6249      NaN           NaN          NaN   
3       3406-8744     NaN           NaN          NaN   
4      9238777734     NaN  9298144-4444          NaN   
5       5018-6500     NaN           NaN          NaN   
6       3601-4461     NaN           NaN  3601-4461     
7       3744-9697     NaN           NaN  3744-9697     
8       3126-7580     NaN           NaN          NaN   
9       4508-9777     NaN           NaN          NaN   

                             Notes   FreeText SalesPersonCode  \
0                            Notes  Free_Text         SlpCode   
1                              NaN        NaN    DAMIÃO SILVA   
2                              NaN        NaN               4   
3     www.portosegimoveis.com.br          NaN               4   
4                              NaN        NaN               7   
5                              NaN        NaN              -1   
6  http://www.g2tecnologia.com.br/        NaN               4   
7  http://www.g2tecnologia.com.br/        NaN               4   
8                              NaN        NaN               4   
9                              NaN        NaN              -1   

                                    EmailAddress ShippingType  \
0                                         E_Mail     ShipType   
1                            cesp@hotmail.com.br          NaN   
2  contato@v4company.com <contato@v4company.com>          NaN   
3                 locacao@portosegimoveis.com.br          NaN   
4                    andreluizalima@hotmail.com           NaN   
5                             edelcy@ntx.com.br           NaN   
6                                            NaN          NaN   
7                                            NaN          NaN   
8                    paulo.vieira@sondait.com.br          NaN   
9                       suporte@mksaude.com.br            NaN   

                Website CompanyPrivate               Unnamed: 16 Unnamed: 17  
0            IntrntSite     CmpPrivate                       NaN         NaN  
1                   NaN       cCompany  EXEMPLO DE PREENCHIMENTO         NaN  
2                   NaN       cCompany                       NaN           C  
3                   NaN       cCompany              

> **Nota:** Decidimos não aplicar a capitalização (conversão de todos os textos para letras maiúsculas) nesta planilha devido à natureza das informações, que envolvem nomes de parceiros de negócios. Como se trata de nomes próprios registrados por empresas, a distinção entre letras maiúsculas e minúsculas é importante para evitar confusões e garantir a correta identificação e referência dos parceiros.

# Tabela CRD1 - Informações de Endereço

A tabela contém as informações de endereço dos parceiros de negócios. Estes dados são aplicáveis tanto para clientes quanto para fornecedores e incluem:

- **Tipo de Endereço:** Identifica se o endereço é de entrega, cobrança, ou outro.

- **Nome do Endereço:** Nome do local ou estabelecimento.

- **Logradouro:** Nome da rua, avenida, ou outro tipo de via.

- **Número, Bairro, Cidade, Estado e CEP:** Detalhes completos do endereço.

- **Complemento:** Informação adicional para localizar o endereço.

Essas informações são usadas para fins de entrega de produtos, correspondência, e pagamentos, assegurando que as operações logísticas ocorram sem falhas.

## Alterações Realizadas

1. **Coluna CardCode:** Remoção de linhas vazias ou que não seguiam as regras de duplas

2. **Coluna Address:** Baseado na coluna AddressType, definimos que todas as linhas com o valor bo_BillTo teriam o valor COBRANCA, e todas as linhas referente a bo_ShipTo teriam o valor DESTINATARIO

3. Padronização das colunas AddrType, Street e StreetNo, dividindo entre tipo de logradouro, endereço e número

- Exemplo: `"Rua Joaquim Floriano, 466"` foi dividido em:
  - `TypeOfAddress`: `"Rua"`
  - `Street`: `"Joaquim Floriano"`
  - `StreetNo`: `"466"`

4. As primeiras linhas da planilha, que continham exemplos de preenchimento, foram removidas.

5. Todos os textos foram convertidos para **maiúsculas**. Além disso, caracteres especiais, como `ç`, foram substituídos por suas versões simplificadas (`c`)

6. Abreviações em `TypeOfAddress` foram substituídas pelas palavras completas:

- Exemplo:
  - `"Av."` foi alterado para `"Avenida"`
  - `"R."` foi alterado para `"Rua"`

7. Os campos de complemento foram mantidos intactos, pois são personalizáveis e opcionais.


8. Correção Manual em `StreetNo`

Linhas específicas receberam atenção manual para corrigir ou ajustar o campo `StreetNo`:

- Linha 102 e 103: `"KM 16"` foi movido para o campo de complemento e substituído por `"S/N"` em `StreetNo`.
- Linha 8: Removida porque o campo estava sem o endereço da rua.
- Linha 724: Removida por ser uma cópia da linha 723, mas sem o CEP.
- Linha 547 e 548, 485 e 486: Inversão corrigida entre `TypeOfAddress` e `StreetNo`.

9. Separação e Correção em `AddressName`

Corrigido `AddressName` que continha endereços completos:

- Exemplo: Linha 277 onde `"RUA HUET BACELAR 306  IPIRANGA  SP"` foi corrigido para `"DESTINATARIO"`.

10. Correções em `StreetNo`
Números em `StreetNo` que terminavam com um espaço extra foram corrigidos:

- Exemplo: `"66 "` foi alterado para `"66"`.

11. Remoção de Linhas com `StreetNo` e Complemento Vazios

Conforme orientação, linhas onde `StreetNo` e o complemento estavam vazios foram removidas.

- Exemplo: Linha 322 foi removida devido ao preenchimento incorreto com `"ADLIABARBOSANEVES395"`.


12. Eliminação de Linhas Duplicadas

Linhas duplicadas, onde apenas o campo de complemento diferia, foram eliminadas.

- Linhas 491 e 492 removidas porque o campo `StreetNo` estava preenchido com `"Trecho III"`.
- Linha 595 removida porque `StreetNo` estava preenchido com o valor `"3,70E+67"`.
- Linha 226 removida porque `StreetNo` estava preenchido com `"215CJ412"`.

13. Correções em `AddressName`
Correções em `AddressName` para padronizar:

- `"Sede Brasil"` foi corrigido para `"Sede"`.

14. Correções de Escrita em Nomes de Ruas

Correções de ortografia em nomes de ruas:

- Exemplo: `"CNDIDO DE ABREU"` foi corrigido para `"CANDIDO DE ABREU"`.

15. Manutenção de Duplicatas com `AddressType` Diferentes

Duplicatas foram mantidas caso tivessem `AddressType` diferentes.

# Tabela CRD7 - Identificação Fiscal


A aba CRD7 concentra os dados de identificação fiscal dos parceiros de negócios, essenciais para o cumprimento das obrigações tributárias e fiscais. Os campos incluídos são:

- **CNPJ/CPF:** Cadastro Nacional de Pessoa Jurídica ou Física.

- **Inscrição Estadual/Municipal:** Registro oficial para fins fiscais e tributários.

- **Código CNAE:** Classificação Nacional de Atividades Econômicas.

- **I.E.S.T:** Inscrição Estadual Substituto Tributário.

- **ID estrangeiro:** Identificação usada para parceiros estrangeiros.

- **Suframa:** Registro para operações na Zona Franca de Manaus.

Essas informações são fundamentais para a emissão de notas fiscais, pagamento de impostos, e outras obrigações fiscais.


## Alterações Realizadas

- Remoção da linha de exemplo de preenchimento

- Alteração das variáveis da coluna `Address` para `ENTREGA` e `COBRANCA`

- Alteração das variáveis da linha `lineNum` para 1 e 2

- Remoção de linhas incorretas

- Remoção das colunas `TaxId5` até `TaxId8`
